modules/contracts/src.ts/services/ethService.ts

```diff
@@ -182,16 +180,18 @@ export class EthereumChainService extends EthereumChainReader implements IVector
 
   /// Helper method to wrap queuing up a transaction and waiting for response.
   private async sendTx(
-    txFn: (gasPrice: BigNumber, nonce?: number) => Promise<TransactionResponse | undefined>,
+    txFn: (gasPrice: BigNumber, nonce: number) => Promise<TransactionResponse | undefined>,
     gasPrice: BigNumber,
+    signer: Signer,
     nonce?: number,
   ): Promise<Result<TransactionResponse | undefined, Error>> {
     // Queue up the execution of the transaction.
     return await this.queue.add(
       async (): Promise<Result<TransactionResponse | undefined, Error>> => {
         try {
           // Send transaction using the passed in callback.
-          const response: TransactionResponse | undefined = await txFn(gasPrice, nonce);
+          const actualNonce: number = nonce ? nonce : await signer.getTransactionCount();
```

Comment:
this will be weird if `nonce` is `0`. that will likely never ever happen in prod, but it's still a better practice to do `nonce ?? await signer.getTransactionCount()`